msteams-platform/graph-api/import-messages/update-import-messages.md

```diff
@@ -0,0 +1,373 @@
+
+```http
+POST https://graph.microsoft.com/beta/teams
+
+Content-Type: application/json
+{
+  "@microsoft.graph.teamCreationMode": "migration",
+  "template@odata.bind": "https://graph.microsoft.com/beta/teamsTemplates('standard')",
+  "displayName": "My Sample Team",
+  "description": "My Sample Team’s Description"
+  "createdDateTime": "2020-03-14T11:22:17.067Z"
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 202 Accepted
+Location: /teams/{teamId}/operations/{operationId}
+Content-Location: /teams/{teamId}
+```
+
+#### Error messages
+
+```http
+400 Bad Request
+```
+
+* `createdDateTime`  set for future.
+* `createdDateTime`  correctly specified, but `teamCreationMode`  instance attribute  is missing or set to invalid value.
+
+## Step Two: Create a channel
+
+Creating a channel for the imported messages is similar to the create team scenario: 
+
+1. [Create a new channel](/graph/api/channel-post?view=graph-rest-beta&tabs=http) with a back-in-time timestamp using the channel resource `createdDateTime` property.
+
+1. Place the new channel in `migration mode`, a special state that bars users from most chat activities within the channel until the migration process is complete.  Include the `channelCreationMode` instance attribute with the `migration` value in the POST request to explicitly identify the new team as being created for migration.  
+<!-- markdownlint-disable MD024 -->
+#### Permissions
+
+|ScopeName|DisplayName|Description|Type|Admin Consent?|Entities/APIs covered|
+|-|-|-|-|-|-|
+|`Teamwork.Migrate.All`|Manage migration to Microsoft Teams|Creating, managing resources for migration to Microsoft Teams|**Application-only**|**Yes**|`POST /teams`|
+
+#### Request (create a channel in migration state)
+
+```http
+POST https://graph.microsoft.com/beta/teams/{id}/channels
+
+Content-Type: application/json
+{
+  "@microsoft.graph.channelCreationMode": "migration",
+  "displayName": "Architecture Discussion",
+  "description": "This channel is where we debate all future architecture plans",
+  "membershipType": "standard",
+  "createdDateTime": "2020-03-14T11:22:17.067Z"
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 202 Accepted
+Location: /teams/{teamId}/channels/{channelId}/operations/{operationId}
+Content-Location: /teams/{teamId}/channels/{channelId}
+```
+
+#### Error message
+
+```http
+400 Bad Request
+```
+
+* `createdDateTime`  set for future.
+* `createdDateTime`  correctly specified but `channelCreationMode`  instance attribute  is missing or set to invalid value.
+
+## Step Three: Import messages
+
+After the team and channel have been created, you can begin sending back-in-time messages using the `createdDateTime`  and `from`  keys in the request body.
+
+#### Request (POST message that is text-only)
+
+```http
+POST https://graph.microsoft.com/beta/teams/teamId/channels/channelId/messages
+
+{
+    "replyToId": null,
+    "messageType": "message",
+    "createdDateTime": "2019-02-04T19:58:15.511Z",
```

Comment:
We would need to update channel.md and team.md documentations to show this new property as well.

We will need to call out the caveat that currently, the createdDateTime field is only getting populated on only those instances of a team or channel that were migrated.

Reply:
channel.md: https://docs.microsoft.com/en-us/graph/api/resources/channel?view=graph-rest-beta

team.md: https://docs.microsoft.com/en-us/graph/api/resources/team?view=graph-rest-beta